### PR TITLE
ocaml-cstruct & ocaml-nocrypto update

### PR DESCRIPTION
**Pull request by sternenseemann**
The trick fixing the build issue was removing `async` support. I am not sure, if this breaks any other packages, but it fixes all the build errors we discovered previously.

cc @vbgl @Zimmi48 

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/cstruct/default.nix [hunk 1]
```diff
@@ -1,20 +1,40 @@
-{stdenv, writeText, fetchurl, ocaml, ocplib-endian, sexplib_p4, findlib,
- async_p4 ? null, lwt ? null, camlp4}:
+{stdenv, buildOcaml, fetchFromGitHub, writeText,
+ ocaml, ocplib-endian, sexplib, findlib, ounit, camlp4,
```
Comment (hannesm) [inline pkgs/development/ocaml-modules/cstruct/default.nix]: as indicated in the other PR, camlp4 can be dropped

File: pkgs/development/ocaml-modules/nocrypto/default.nix [hunk 2]
```diff
@@ -1,28 +1,37 @@
-{ stdenv, fetchzip, ocaml, findlib, cstruct, type_conv, zarith, ounit }:
+{ stdenv, buildOcaml, fetchFromGitHub, ocaml, findlib, cstruct, type_conv, zarith, ounit, ocaml_oasis, ppx_sexp_conv
```
Comment (hannesm) [inline pkgs/development/ocaml-modules/nocrypto/default.nix]: type_conv can be dropped (it was there to get a sexplib with camlp4 support, which is no longer needed)... but sexplib is needed for nocrypto (which might be there due to transitive dependencies (i.e. cstruct depends on sexplib already))

**Comment by sternenseemann** [comment]
Oops, this is a semi-duplicate PR I accidentaly opened…

File: pkgs/development/ocaml-modules/cstruct/default.nix (hunk 1 above)
Comment (sternenseemann) [inline pkgs/development/ocaml-modules/cstruct/default.nix]: Yeah, I fixed it in the other PR, somehow I pull-requested the cstruct-update twice.

File: pkgs/development/ocaml-modules/nocrypto/default.nix (hunk 2 above)
Comment (sternenseemann) [inline pkgs/development/ocaml-modules/nocrypto/default.nix]: Thank you, will remove!

**Pull request closed without merging**